subo/release/version.go

```diff
@@ -12,3 +21,22 @@ var AtmoVersion = "0.4.2"
 
 // SCCTag is the docker tag used for creating new compute core deployments
 var SCCTag = "v0.1.0"
+
+// CheckForLatestVersion returns an error if SuboDotVersion does not match the latest GitHub release or if the check fails
+func CheckForLatestVersion() error {
+	latestRepoRelease, _, err := github.NewClient(nil).Repositories.GetLatestRelease(context.Background(), "suborbital", "subo")
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch latest subo release")
+	}
+	latestCmdVersion, err := version.NewVersion(*latestRepoRelease.TagName)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse latest subo version")
+	}
+	cmdVersion, err := version.NewVersion(SuboDotVersion)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse current subo version")
+	} else if cmdVersion.LessThan(latestCmdVersion) {
```

Comment:
I'm wondering if we should only print the warning when it's a new minor version? I wouldn't want to bug them for every patch version... what do you think @jagger27 ?

Reply:
I was working on this myself last week and I have a few different thoughts on it. 

We probably shouldn't be hammering the GitHub API every time subo runs. The check itself (as written above) takes on the order of 300ms on my connection and machine, which isn't great. I think we should also avoid doing an update check when subo is running in Docker, since it's not relevant there. I was thinking to `touch` a "last checked for updates" file in `~/.local/share/suborbital` and use that timestamp to check for updates less often.

Another idea I had was to defer the update notification to the termination of the program and spawn a goroutine to do the check in the background. 

I think using `util.LogInfo` would be more appropriate here as well, unless we start talking about patch severity like what `npm` does with some colourized output (https://github.com/fatih/color). It would also be useful to detect how subo was installed (Homebrew or not) and provide appropriate update instructions as well.

All of that is getting ahead of ourselves a bit, of course. :^)

```
⚠️ WARNING: upgrade subo 0.2.0 to the latest release 0.2.1
```
to
```
ℹ️  An upgrade for subo is available: v0.2.0 → v0.2.1
```

Personally, I want to be notified for 0.0.X releases, but I could see that getting annoying.

Other feedback, the update checker shouldn't consider an available update to be an error. So let's add a `bool` or maybe just the new version number as a `string` to the output of `CheckForLatestVersion()` along with any errors.

Reply:
@jagger27 thx for the feedback! Tbh, I like both of your ideas to check a timestamp and to defer the check. If we don't defer the check, it'll be easy to miss the message.

So here's my notes so far:

1. Cache the latest version check timestamp in somewhere like `~/.local/share/suborbital` and use that timestamp to determine whether or not to check remotely again.
2. Defer the check
3. Switch from logging a warning to info
4. Update `CheckForLatestVersion()` output

Reply:
Yep, that's all good @denopink! Thanks for working on this.

It would also make sense to put all of the update related code in its own file (it can stay in the release package). `version.go` should stay as it was.

Also, make sure you're using the latest version of github.com/google/go-github.

Reply:
Cool, just updated the code based on the feedback.